type-definitions/immutable.js.flow

```diff
@@ -561,9 +561,78 @@ declare module 'immutable' {
     flatten(shallow?: boolean): /*this*/Map<any,any>;
   }
 
-  // OrderedMaps have nothing that Maps do not have. We do not need to override constructor & other statics
-  declare class OrderedMap<K,V> extends Map<K,V> {
+  declare class OrderedMap<K,V> extends KeyedCollection<K,V> {
```

Comment:
I think it's because of `this` in the `update` definition. Does it happen if you replace `this` on line 574 with `OrderedMap<K,V>` and with `Map<K,V> on line 580?